dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/acl/DefaultTrackerOwnershipManager.java

```diff
@@ -149,72 +149,41 @@ public void transferOwnership(
 
   @Override
   @Transactional
-  public void assignOwnership(
-      TrackedEntity trackedEntity,
-      Program program,
-      OrganisationUnit organisationUnit,
-      boolean skipAccessValidation,
-      boolean overwriteIfExists) {
-    if (trackedEntity == null || program == null || organisationUnit == null) {
-      return;
+  public void grantTemporaryOwnership(
+      @Nonnull TrackedEntity trackedEntity, Program program, UserDetails user, String reason)
+      throws ForbiddenException {
+    if (canSkipOwnershipCheck(user, program)) {
+      throw new ForbiddenException(
```

Comment:
Doesn't this break the API? 
I feel it may be intrusive to throw an exception for the scenario were the check itself could be skipped? Thoughts?

Reply:
The issue was that the check was used to skip the actual action that is granting temporary access.
In my opinion the API was just wrong because when the user was asking access for a not existent program, the API would return with a positive answer even though nothing happened.

Reply:
I agree with the program non-existent and non-tracker program part. 
For superuser, I am a bit concerned if it will break the UI workflow in some way. But maybe it should still not end up here anyway, since they are a superuser 🤔

Reply:
That was my thinking. In our current flow, a superuser will not end up here since they have access to every TE in the system, right?
It's my understanding that a superuser will always have capture scope access to all org units in the implementation, so the capture app would never send such a user to this endpoint.
If this assumption is incorrect, should we consider granting temporary access to the superuser as well?

Reply:
I guess the assumption is correct. Let us see how this goes.